StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1509UnitTests.cs

```diff
@@ -0,0 +1,702 @@
+        [TestMethod]
+        public async Task TestEmptySource()
+        {
+            var testCode = string.Empty;
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [TestMethod]
+        public async Task TestClassDeclarationOpeningBraceHasBlankLine()
+        {
+            var testCode = @"
+class Foo
+
+{
+
+}";
+
+            var expected = new[]
+                {
+                    new DiagnosticResult
+                    {
+                        Id = this.DiagnosticId,
+                        Message = "Opening curly brackets must not be preceded by blank line.",
+                        Severity = DiagnosticSeverity.Warning,
+                        Locations =
+                            new[]
+                            {
+                                new DiagnosticResultLocation("Test0.cs", 4, 1)
+                            }
+                    }
+                };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None);
+        }
+
+        [TestMethod]
+        public async Task TestClassDeclarationOpeningBraceDoesntHaveBlankLine()
+        {
+            var testCode = @"
+class Foo
+{
+
+}";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [TestMethod]
+        public async Task TestClassDeclarationCommentBeforeOpeningBrace()
+        {
+            var testCode = @"
+class Foo
+//this is a comment
+{
+
+}";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [TestMethod]
+        public async Task TestClassDeclarationMultilineCommentBeforeOpeningBrace()
+        {
+            var testCode = @"
+class Foo
+/*this is a comment
+that spans 2 lines */
+{
+
+}";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [TestMethod]
+        public async Task TestClassDeclarationMultilineCommentBeforeOpeningBraceButBlankLineExists()
+        {
+            var testCode = @"
+class Foo
+/*this is a comment
+that spans 2 lines */
+
+//another comment
+{
+
+}";
+
+            var expected = new[]
+                {
+                    new DiagnosticResult
+                    {
+                        Id = this.DiagnosticId,
+                        Message = "Opening curly brackets must not be preceded by blank line.",
+                        Severity = DiagnosticSeverity.Warning,
+                        Locations =
+                            new[]
+                            {
+                                new DiagnosticResultLocation("Test0.cs", 7, 1)
+                            }
+                    }
+                };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None);
+        }
+
```

Comment:
We don't do everything the same way StyleCop does. We have some differences in places where we think that StyleCop doesn't do the right thing. But I'm not sure whats the best thing to do here.